util/src/keys/directory.rs

```diff
@@ -484,6 +484,15 @@ impl KeyDirectory {
 			let json_bytes = json_text.into_bytes();
 			try!(file.write(&json_bytes));
 		}
+		{
+			let cstr = ::std::ffi::CString::new(self.key_path(&key_file.id).as_path().to_str().unwrap())
+				.unwrap();
+			let result_i32 = unsafe { ::libc::chmod(cstr.as_ptr(), ::libc::S_IWUSR | ::libc::S_IRUSR) };
+			if result_i32 != 0 {
+				fs::remove_file(self.key_path(&key_file.id)).unwrap();
+				panic!("fatal: failed to modify permissions of the file (chmod: {})", result_i32);
+			}
+		}
```

Comment:
Can you move it to a separate function? I believe that it might be useful in the other places. Also it will be easier to write a windows implementation.